### ISSUE TITLE
Consider removing Rackspace support

### ISSUE THREAD

**Issue by erictune** [kubernetes/kubernetes#6962]
Rackspace support has not ben updated in a few months.
cluster/rackspace/cloud-config/minion-cloud-config.yaml is using etcd_servers for kubelet and kube-proxy, but support for them has been removed since v0.9, I think.

I'll remove it in a few days if no-one makes a counter-offer to get the config working with latest release.

**Comment by thockin** [comment]
There's been some work on this but it is blocked while we figure out some
load-balancer design stuff.

**Comment by erictune** [comment]
Okay, I will remove the link to rackspace until it is fixed (#6987).

Leaving this issue open to remind us to delete it in, say, a month, if no progress is made.

**Comment by bgrant0607** [comment]
cc @doublerr

**Comment by bgrant0607** [comment]
See also #5215

**Comment by doublerr** [comment]
I'd prefer to try to get things updated and keep support. However I don't want to continually become stale.  every few months either.

**Comment by fxpz** [comment]
Hi all,

I want to use kubernetes on my existing rackspace infrastructure but I don't understand if rackspace suppor was removed or not. In Doc, under "custom cloud solution" it's linked but I wasn't able to deploy it neither with `get.k8s.io` install method nor with `hack/dev-build-and-up.sh`. With `dev-build-and-up.sh` i got a `cannot allocate memory` error (I have 16GB or RAM!).

thanks in advance,
Fx

**Comment by erictune** [comment]
It looks like no support has happened on Rackspace for a year.

**Comment by erictune** [comment]
You should not expect it to work, and we should delete rackspace support.

**Comment by thockin** [comment]
ACK.

**Comment by doublerr** [comment]
Unfortunately I think we'll need to drop support for Rackspace as a deployment target since I can't commit to keeping the deployment scripts up to date. 

Rackspace support as a _cloudprovider_ should be maintained.

**Comment by devx** [comment]
I would be willing to support the Rackspace deployment target, if there is interest to keep it alive.

**Comment by w0mbat** [comment]
@devx There would be definitely an interest from our side - sad to see that Rackspace has no interest in maintaining it

**Comment by devx** [comment]
No problem, I will take over the maintenance. It will take me a bit of time to get caught up, but I will get it done as soon as possible.

**Comment by w0mbat** [comment]
@devx Do you think it would be hard to get the [OpenStack Implementation] of LBaaS (https://github.com/kubernetes/kubernetes/blob/4694a6dd7134135ce249ec403882e01a54712c4a/pkg/cloudprovider/providers/openstack/openstack.go#L509) working [here](https://github.com/kubernetes/kubernetes/blob/4694a6dd7134135ce249ec403882e01a54712c4a/pkg/cloudprovider/providers/rackspace/rackspace.go#L456) ?

**Comment by devx** [comment]
@w0mbat Sorry it took me a bit to respond. there is a quick update as to where I am.

I was busy with some stuff, that prevented me from working on this. But that has been resolved, and should have updates by the end of next week. However what's causing the delays is the lack of a good API for Rackspace Cloud.  I've used the gophercloud, it works but I'm not a fan.  Additionally gophercloud doesn't have full API support (one of the things I'm not a fan).  This would make it hard to implement LBaaS.  I should have a better idea of your request sometime next week.
